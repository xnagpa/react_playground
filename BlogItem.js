const {
  bind
} = _;

class Image extends React.Component {
  render() {
    return React.createElement('img', {
      src: this.props.src
    })
  }
}


Image.propTypes = {
  src: React.PropTypes.string
}

Image.defaultProps = {
  src: ''
}


class TextBox extends React.Component {
  render() {
    return React.createElement(Span, null, this.props.children)
  }
}


class Span extends React.Component {
  render() {
    return React.createElement('p', {
        style: {
          color: 'red'
        }
      },
      this.props.children)
  }
}

Span.propTypes = {
  children: React.PropTypes.string
}

Span.defaultProps = {
  children: "Hello Jake!"
}

class Like extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      count: props.count
    }
    this.handleClick = bind(this.handleClick, this);
  }

  handleClick() {
    this.setState({
      count: this.state.count + 1
    })
  }

  render() {
    return ( <div><p> {
        this.state.count
      }
      likes </p><button onClick = {
      this.handleClick
    }> Like this shit </button></div>
  )
}

}

class MetaData extends React.Component {
  render() {
    const {
      author,
      updatedAt,
      createdAt
    } = this.props;
    return (React.createElement('div', {},
      author && React.createElement(TextBox, {}, `${author.name}, ${author.age}`),
      React.createElement(TextBox, {}, `Отредактировано ${updatedAt}`),
      React.createElement(TextBox, {}, `Создано ${createdAt}`),
    ))
  }
}

MetaData.defaultProps = {
  author: {
    name: '',
    age: ''
  },
  updatedAt: '',
  createdAt: ''
}

MetaData.propTypes = {
  author: React.PropTypes.shape({
    name: React.PropTypes.string,
    age: React.PropTypes.string
  }),
  updatedAt: React.PropTypes.string,
  createdAt: React.PropTypes.string
}

class BlogItem extends React.Component {
  render() {
    const {
      meta,
      image,
      text,
      likes
    } = this.props;
    return React.createElement('div', {},
      React.createElement(Image, {
        src: image.url
      }),
      React.createElement(TextBox, {}, text),
      React.createElement(Like, {
        count: likes
      }),
      meta && React.createElement(MetaData, {
        author: meta.author,
        updatedAt: meta.updatedAt,
        createdAt: meta.createdAt
      })
    )
  }
}

BlogItem.defaultProps = {
  author: {
    name: '',
    age: ''
  },
  text: '',
  image: {
    width: 0,
    height: 0,
    url: ''
  },
  updatedAt: '',
  createdAt: '',
  likes: 0
}

BlogItem.propTypes = {
  meta: MetaData.propTypes,
  text: React.PropTypes.string,
  image: Image.propTypes,
  likes: React.PropTypes.number
}



ReactDOM.render(
  React.createElement('div', {}, React.createElement(BlogItem, {
    meta: {
      author: {
        name: "Finn",
        age: "12"
      },
      updatedAt: moment().subtract(10, 'days').calendar(),
      createdAt: moment().subtract(10, 'days').calendar()
    },
    likes: 651,
    text: 'Fun will never end',
    image: {
      width: 100,
      height: 100,
      url: 'http://i.cdn.turner.com/v5cache/CARTOON/site/Images/i70/adventure-time.png'
    }
  })),
  document.getElementById("app")
);
