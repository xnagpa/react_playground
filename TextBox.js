const DOM = React.DOM;
const TextBox = (props) => (
  React.createElement(
    Span,
    {str: props.str}
  )
);

const Span = (props) => (
  DOM.span({
    style: {
      border: '1px solid red'
    }
  }, props.children, props.str)
)

ReactDOM.render(
  React.createElement(
    TextBox, {str: "Hoola hoop"}),
  document.getElementById("app")
);
