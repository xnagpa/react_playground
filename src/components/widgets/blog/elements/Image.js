import React from 'react';
import { Image as SemanticImage } from 'semantic-ui-react';
import PropTypes from 'prop-types';

const Image =  (props) => (
  <SemanticImage src={props.src}/>
);


Image.propTypes = {
  src: PropTypes.string
};

Image.defaultProps = {
  src: ''
};

export default Image;
