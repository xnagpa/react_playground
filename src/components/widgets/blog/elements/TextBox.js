import React from 'react';

const TextBox =  (props) => (
  <Span>{ props.children }</Span>
);

TextBox.propTypes = {
  children: React.PropTypes.string
};


const Span =  (props) => (
  <p style={{
    color: 'red'
  }}>{props.children}</p>
);

Span.propTypes = {
  children: React.PropTypes.string
};

Span.defaultProps = {
  children: 'Hello Jake!'
};

export default TextBox;
