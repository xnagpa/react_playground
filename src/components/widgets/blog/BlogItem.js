import React from 'react';
import Image from './elements/Image';
import TextBox from './elements/TextBox';
// import Like from './elements/Like';
import MetaData from './elements/MetaData';
import Link from 'components/elements/Link';
import { postsPath } from 'helpers/routes';
import { Item } from 'semantic-ui-react';
import LikeContainer from 'containers/LikeContainer';
import PropTypes from 'prop-types';

const BlogItem =  (props) => {
  const {
    editPath,
    meta,
    image,
    text,
    id
  } = props;

  return <Item>
    { image && <Item.Image size='medium' src={ image.src } />}
    <Item.Content>
      <Item.Header>
        <Link to={ postsPath(id) }>{ text }</Link>
      </Item.Header>
      <Item.Description>
        <Image
          src='http://semantic-ui.com/images/wireframe/short-paragraph.png'/>
        <TextBox>{ text }</TextBox>
      </Item.Description>
      <Item.Meta>
        { meta && <MetaData author={meta.author}
        updatedAt= { meta.updatedAt }
        createdAt= {meta.createdAt}/> }
      </Item.Meta>
      <Item.Extra>
        <LikeContainer id={ id }/>
      </Item.Extra>
      <Item.Extra>
        <Link to={ editPath }>Edit this post</Link>
      </Item.Extra>
    </Item.Content>
  </Item>;
};

BlogItem.defaultProps = {
  text: '',
  likes: 0,
  id: 0
};

BlogItem.propTypes = {
  meta: PropTypes.shape(MetaData.propTypes),
  text: PropTypes.string,
  image: PropTypes.shape(Image.propTypes),
  likes: PropTypes.number,
  increaseLikesHandler: PropTypes.func,
  id: PropTypes.number
};

export default BlogItem;
