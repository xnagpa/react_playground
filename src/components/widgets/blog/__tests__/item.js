import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Link from 'components/elements/Link';
import BlogItem from '../BlogItem';
import TextBox from '../elements/TextBox';
import Image from '../elements/Image';
import { Provider } from 'react-redux';

const store = {
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({
    posts: {
      isFetching: false,
      error: false,
      entries: [{
        id: 0,
        meta: {},
        image: {},
        text: 'Finn and Jake`s scrambled eggs',
        likes: 60,
        totalCount: 1,
        editPath: '/posts/1/edit'
      }]
    },
    post: {
      isUpdating: false,
      isFetching: false,
      error:false,
      entry:null
    },
    comments: {
      isUpdating: false,
      isFetching: false,
      error:false,
      entry:null
    },
    like: {
      isFetching: false,
      error: false,
      entries: []
    }
  }),
};


describe('BlogItem', () => {
  it('renders without crashing', () => {
    const div =  document.createElement('div');
    ReactDOM.render(<Provider store={store}><BlogItem/></Provider>, div);
  });
});

describe('render', () => {
  it('renders the title', () => {
    const itemProps = {
      text: "Finn and Jake`s scrambled eggs",
      id: 5,
      image: {
        src: 'http://semantic-ui.com/images/wireframe/short-paragraph.png'
      }
    };
    const item = shallow(<BlogItem {...itemProps}/>);
    const text = <TextBox>{ "Finn and Jake`s scrambled eggs" }</TextBox>;
    const image = <Image
      src='http://semantic-ui.com/images/wireframe/short-paragraph.png'/>;

    expect(item.contains(text)).toEqual(true);
    expect(item.contains(image)).toEqual(true);
  });

  it('renders common item', () => {
    const itemProps = {
      text: "Finn and Jake`s scrambled eggs",
      id: 5,
      image: {
        src: 'http://semantic-ui.com/images/wireframe/short-paragraph.png'
      }

    };
    const item = shallow(<BlogItem {...itemProps}/>);
    expect(item).toMatchSnapshot();
  });
});
