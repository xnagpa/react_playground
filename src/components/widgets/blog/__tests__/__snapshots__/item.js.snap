// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders common item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <ItemImage
        size="medium"
        src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
    />
    <ItemContent>
        <ItemHeader>
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/5"
            >
                Finn and Jake\`s scrambled eggs
            </Link>
        </ItemHeader>
        <ItemDescription>
            <Image
                src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
            />
            <TextBox>
                Finn and Jake\`s scrambled eggs
            </TextBox>
        </ItemDescription>
        <ItemMeta />
        <ItemExtra>
            <Connect(Like)
                id={5}
            />
        </ItemExtra>
        <ItemExtra>
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={undefined}
            >
                Edit this post
            </Link>
        </ItemExtra>
    </ItemContent>
</Item>,
  "nodes": Array [
    <Item>
      <ItemImage
            size="medium"
            src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
      />
      <ItemContent>
            <ItemHeader>
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/5"
                  >
                        Finn and Jake\`s scrambled eggs
                  </Link>
            </ItemHeader>
            <ItemDescription>
                  <Image
                        src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
                  />
                  <TextBox>
                        Finn and Jake\`s scrambled eggs
                  </TextBox>
            </ItemDescription>
            <ItemMeta />
            <ItemExtra>
                  <Connect(Like)
                        id={5}
                  />
            </ItemExtra>
            <ItemExtra>
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to={undefined}
                  >
                        Edit this post
                  </Link>
            </ItemExtra>
      </ItemContent>
</Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        id={5}
        image={
                Object {
                        "src": "http://semantic-ui.com/images/wireframe/short-paragraph.png",
                      }
        }
        likes={0}
        text="Finn and Jake\`s scrambled eggs"
/>,
      "_debugID": 35,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "id": 5,
          "image": Object {
            "src": "http://semantic-ui.com/images/wireframe/short-paragraph.png",
          },
          "likes": 0,
          "text": "Finn and Jake\`s scrambled eggs",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 21,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <ItemImage
                    size="medium"
                    src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Finn and Jake\`s scrambled eggs
                              </Link>
                    </ItemHeader>
                    <ItemDescription>
                              <Image
                                        src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
                              />
                              <TextBox>
                                        Finn and Jake\`s scrambled eggs
                              </TextBox>
                    </ItemDescription>
                    <ItemMeta />
                    <ItemExtra>
                              <Connect(Like)
                                        id={5}
                              />
                    </ItemExtra>
                    <ItemExtra>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to={undefined}
                              >
                                        Edit this post
                              </Link>
                    </ItemExtra>
          </ItemContent>
</Item>,
        "_debugID": 36,
        "_renderedOutput": <Item>
          <ItemImage
                    size="medium"
                    src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Finn and Jake\`s scrambled eggs
                              </Link>
                    </ItemHeader>
                    <ItemDescription>
                              <Image
                                        src="http://semantic-ui.com/images/wireframe/short-paragraph.png"
                              />
                              <TextBox>
                                        Finn and Jake\`s scrambled eggs
                              </TextBox>
                    </ItemDescription>
                    <ItemMeta />
                    <ItemExtra>
                              <Connect(Like)
                                        id={5}
                              />
                    </ItemExtra>
                    <ItemExtra>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to={undefined}
                              >
                                        Edit this post
                              </Link>
                    </ItemExtra>
          </ItemContent>
</Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    id={5}
    image={
        Object {
            "src": "http://semantic-ui.com/images/wireframe/short-paragraph.png",
          }
    }
    likes={0}
    text="Finn and Jake\`s scrambled eggs"
/>,
}
`;
